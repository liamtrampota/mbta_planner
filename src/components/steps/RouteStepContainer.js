import React from 'react';
import '../../styles/StepContainer.css';

const RouteStepContainer = ({ selectRoute, selectedRoute, routes }) => (
  <div>
    <h2 style={{ textAlign: 'center' }}>
      Select your route
    </h2>
    { routes ?
      <RouteStep routes={routes} selectRoute={selectRoute} selectedRoute={selectedRoute} />
      :
      <h4>
        Loading
      </h4>
    }
  </div>
)

const RouteStep = ({ routes, selectRoute, selectedRoute }) => (
  <div className="stepContainer">
    {routes.map(route => <Route key={route.id} route={route} selectRoute={selectRoute} selectedRoute={selectedRoute} />)}
  </div>
)

const Route = ({ route, selectRoute, selectedRoute }) => (
  <button style={{ display: 'flex', alignItems: 'center', margin: '5px', minWidth: '150px', backgroundColor: selectedRoute && selectedRoute.id===route.id ? 'lightgreen' : '' }} onClick={()=>selectRoute(route)}>
    <div style={{ height: '10px', width: '10px', marginRight: '10px', backgroundColor: `#${route.attributes.color}`}}/>
    <h4>
      {route.attributes.long_name}
    </h4>
  </button>
)

export default RouteStepContainer;
