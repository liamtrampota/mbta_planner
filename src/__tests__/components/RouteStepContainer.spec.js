import React from 'react';
import { shallow } from 'enzyme';
import RouteStepContainer from '../../components/steps/RouteStepContainer';

const props = {
  selectRoute: jest.fn(),
  selectedRoute: { attributes: { direction_names: ["South", "North"], direction_destinations: ["Ashmont", "Montclair"] }},
  routes: [{ id: '1', attributes: { long_name: 'Red'} }]
}

describe('RouteStepContainer', () => {
  test('renders RouteStepContainer', () => {
    const wrapper = shallow(<RouteStepContainer {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
