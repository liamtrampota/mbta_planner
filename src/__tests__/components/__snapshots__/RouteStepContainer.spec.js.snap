// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteStepContainer renders RouteStepContainer 1`] = `ShallowWrapper {}`;
